### PR TITLE
Fix the (object detail) flake for repro 39477

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### What does this PR accomplish?
This PR fixes the E2E flake (repro for #39477).

The example of a previous failure:
https://github.com/metabase/metabase/actions/runs/10055141486/job/27792441862#step:11:471
scenarios  question  object details -- shows correct object detail card for questions with joins after clicking on view details (metabase#39477) (failed) (attempt 2)

### Explanation
It misclicked.
It was supposed to click on the third row, but it clicked on the fourth one instead.

I am not entirely sure what's the root cause, but we're using a more specific selector in this updated test version. It should make sure that we click on an object detail trigger for that specific row.

### Stress-test
50/50 ✅ https://github.com/metabase/metabase/actions/runs/10057774847

### CODE REVIEW

File: e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js [hunk 1]
```diff
@@ -101,30 +101,18 @@ describe("scenarios > question > object details", { tags: "@slow" }, () => {
...
-      cy.get("h2").should("contain", "Order").should("contain", 1);
-      cy.findByTestId("object-detail-close-button").click();
-    });
-
-    cy.log("check click on 3rd row");
-
-    cy.get("[data-testid=cell-data]").contains("52.72").realHover();
-    cy.findByTestId("detail-shortcut").findByRole("button").click();
-
+    cy.log("Check object details for the first row");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js:111]: We need to open the object detail modal at least twice, see repro steps for #39477

**Review by kamilmielnik — CHANGES_REQUESTED** [review]

File: e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js:111]: In that case we might have a regression, and not a flake.
It's really hard to determine whether this test fails or flakes.

I'll give it another try.

File: e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js [hunk 2]
```diff
@@ -101,30 +101,28 @@ describe("scenarios > question > object details", { tags: "@slow" }, () => {
...
+    cy.findByTestId("question-row-count").should("have.text", "Showing 2 rows");
 
-    cy.log("check click on 1st row");
-
-    cy.get("[data-testid=cell-data]").contains("37.65").realHover();
-    cy.findByTestId("detail-shortcut").findByRole("button").click();
-
+    cy.log("Check object details for the first row");
+    cy.findAllByTestId("cell-data").filter(":contains(37.65)").realHover();
+    cy.get("[data-show-detail-rowindex='0']").click();
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js:113]: I find this selector to be much more reliable and explicit than targeting the test-id that's the same for any row.

File: e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js [hunk 3]
```diff
@@ -101,30 +101,28 @@ describe("scenarios > question > object details", { tags: "@slow" }, () => {
             alias: "Products",
           },
         ],
-        limit: 5,
+        limit: 2,
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js:104]: We don't have a way to scope the search for the table cell within a single row (see this thread). We have to grab ALL table cells like this `cy.findAllByTestId("cell-data")`. That includes the header table cells btw.

So it makes sense to limit the number of total rows (the number of total cells) for the performance reasons.

File: e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/visualizations-tabular/object_detail.cy.spec.js:111]: Updated the test and re-ran the stress-test.
Updated the PR description as well.
https://github.com/metabase/metabase/pull/45973/commits/369214007642a275d133f75c2149ef4d83e89d1a

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**